### PR TITLE
Attach muon plotting to GUI

### LINKED ISSUE #26883: Attach plotting to muon GUI's
Want to attache the current muon plotting to the muon analysis and frequency domain analysis GUI's

This can be done as a dock widget

### PR DESCRIPTION
**Pull request by StephenSmith25**
**Description of work.**
This PR improves the plotting within the Muon analysis and Frequency domain analysis GUIs by making the plotting window a dockable widget. This change is made to improve the user experience, particularly when working on laptops. This change is implemented through a QtDockWidget.

**To test:**
(1) Load up the Muon analysis GUI
(2) Load a run data (e.g 2223)
(3) Check that plot is created
(4) Change group / pair to bwd, check that the plot updates correctly
(5) Change plot type to counts and check that the plot updates correctly
(6) Move the new dock widget about, ensuring that no issues are encountered.

Fixes #26883. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/release/v4.3.0/muon.rst [hunk 1]
```diff
@@ -5,9 +5,21 @@ MuSR Changes
...
-.. warning:: **Developers:** Sort changes under appropriate heading
-    putting new features at the top of the section, followed by
-    improvements, followed by bug fixes.
+New
+###
 
+Improvements
+############
+
+Algorithms
```
Comment (AnthonyLim23) [inline docs/source/release/v4.3.0/muon.rst]: this is an interface change not an algorithm

File: scripts/Muon/GUI/MuonAnalysis/muon_analysis_2.py [hunk 2]
```diff
@@ -62,6 +63,7 @@ def __init__(self, parent=None):
         except AttributeError as error:
             self.warning_popup(error.args[0])
 
+        # self.setAttribute(QtCore.Qt.WA_DeleteOnClose)
```
Comment (ConorMFinn) [inline scripts/Muon/GUI/MuonAnalysis/muon_analysis_2.py]: Is this being kept for a reason?

**Review by cailafinn — CHANGES_REQUESTED** [review]
Just the one code comment. 

I did find when doing functional testing that mantid will soft crash with an unhandled `KeyError: "'EMU2233; Pair Asym; long; MA' does not exist.` exception, if a workspace is loaded by the GUI, it is then deleted, and you attempt to load a different one.

File: scripts/Muon/GUI/MuonAnalysis/muon_analysis_2.py (hunk 2 above)
Comment (StephenSmith25) [reply scripts/Muon/GUI/MuonAnalysis/muon_analysis_2.py]: Good spot, this was left in by accident.

**Comment by StephenSmith25** [comment]
> Just the one code comment.
> 
> I did find when doing functional testing that mantid will soft crash with an unhandled `KeyError: "'EMU2233; Pair Asym; long; MA' does not exist.` exception, if a workspace is loaded by the GUI, it is then deleted, and you attempt to load a different one.

I've added a function to make sure the plot window updates when the underlying workspace is deleted. This should address the issue you found.

**Review by cailafinn — CHANGES_REQUESTED** [review]
Good fix for the deleted workspaces. 

While testing I noticed that changing the steps value in the Rebin field does not update the plot unless you make some other other change, such as ticking and unticking "plot raw" or changing the "group/pair" setting.

**Comment by StephenSmith25** [comment]
> Good fix for the deleted workspaces.
> 
> While testing I noticed that changing the steps value in the Rebin field does not update the plot unless you make some other other change, such as ticking and unticking "plot raw" or changing the "group/pair" setting.

Good spot again. The graph should now update if you type a step value in the rebin field and press enter.

**Review by cailafinn — DISMISSED** [review]
This looks good to me now. Good work :+1:

**Comment by AnthonyLim23** [comment]
It only makes sense to update the plot if use raw data is unticked

**Comment by AnthonyLim23** [comment]
@ConorMFinn are you happy with this PR?

**Review by cailafinn — CHANGES_REQUESTED** [review]
Functions well and code looks good. Just needs the merge conflicts solving and it can be approved.

**Review by cailafinn — APPROVED** [review]
Looks good to me now. Good work :+1:

**Pull request merged**